CRM/Core/Payment/OmnipayMultiProcessor.php

```diff
@@ -601,6 +639,42 @@ private function getCorePaymentFields() {
   }
 
   /**
+   * Get payment processor payment fields.
+   *
+   * @return array
+   */
+  private function getProcessorPaymentFields() {
```

Comment:
OK, now I see what you are doing here. There is no metadata for the field 'cardholder_name', or for that matter 'card_expiry_month' or 'card_expiry_year' so you are adding it.

I think we can accept there are some generic fields which the metadata is not provided for in the parent function getPaymentFormFieldsMetadata()

I think a better way to do this would be to override that function ie

function getPaymentFormFieldsMetadata() {
 $metadata = parent::getPaymentFormFieldsMetadata();
  $metadata['cardholder_name'] => array(

 ),
  $metadata['expiry_month'] => array(

 ),
  $metadata['expiry_year'] => array(

 ),

These fields are like a library of generic fields to choose from.

Then the getPaymentFieldMapping() function would be the only function that would need to be payment processor specifically hacked, to specific which fields to use from that library.
}